In [41]:
import numpy as np
from scipy.stats import t,norm

In [42]:
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32,
                 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
n=len(data)
mean=np.mean(data)
s=np.std(data,ddof=1)
alpha=0.01
alpha2=alpha/2
df=n-1

t-test

In [50]:
t_critical = t.ppf(1 - alpha/2, df)
margin_error_t = t_critical * (s/ np.sqrt(n))
ci_t = (mean - margin_error_t, mean + margin_error_t)


In [ ]:
# We use the t-distribution because:
# 1. The population standard deviation (σ) is unknown.
# 2. The sample size is small (n = 15).
# 3. The t-distribution accounts for extra uncertainty by having wider tails.
# 4. As n increases, t-distribution approaches the normal (z) distribution.


z-test

In [46]:
p_std_dev = 0.2
z_critical =norm.ppf(1 - alpha/2)
margin_error_z = z_critical * (p_std_dev / np.sqrt(n))
ci_z = (mean - margin_error_z, mean + margin_error_z)



In [55]:
print(f"Sample Mean: {mean:.4f}")
print(f"Sample sd  = {s:.6f}")
print(f"99% CI using sample std dev (t): ({ci_t[0]:.4f}, {ci_t[1]:.4f})")
print(f"99% CI using known pop std dev (z): ({ci_z[0]:.4f}, {ci_z[1]:.4f})")

Sample Mean: 1.2387
Sample sd  = 0.193164
99% CI using sample std dev (t): (1.0902, 1.3871)
99% CI using known pop std dev (z): (1.1057, 1.3717)


The t-interval is wider (1.090 to 1.387) because we used s (estimated σ) → more uncertainty.

The z-interval is slightly narrower (1.106 to 1.372) since it assumes the true σ is known (= 0.2).